tests/diag.sh

```diff
@@ -880,6 +893,87 @@ wait_queueempty() {
 	error_exit $TB_ERR_TIMEOUT
 }
 
+# The pidfile is one way of showing the process id of the
+# rsyslogd process, but fuser can tell us the pid of the
+# process that is actually running by looking at the pid
+# file lock barrier, which rsyslogd locks with fcntl()
+# and holds open for the lifetime of the process.
+#
+# This validation is important because the pidfile could
+# have been written by an instance that crashed, and another
+# process could have gotten the same pid, but linux guarantees
+# that the flock will be cleared when the process is dead,
+# even though the file itself may have been created earlier.
+#
+# $1 is the instance
+# $2 abort on invalid result
+validate_pidfile() {
+
+	RETCODE=0
+	ABORT_ON_INVALID=1
+	pidfile_pid=""
+	fuser_pid=""
+
+	if [ "$2" = "no" ]; then
+		ABORT_ON_INVALID=0
+	fi
+
+	PF=$RSYSLOG_PIDBASE$1.pid
+	if [ -z $PF ]; then
+		echo "validate_pidfile: \$RSYSLOG_PIDBASE is not set!"
+		return 1
+	fi
+
+	if [ ! -f $PF ]; then
+		echo "validate_pidfile: the file '$PF' is missing"
+		RETCODE=1
+	else
+		pidfile_pid=$(cat "$PF")
+		if [ -z $pidfile_pid ]; then
+			echo "validate_pidfile: pidfile pid was not detected"
+			RETCODE=1
+		fi
+	fi
+
+	if [ -f ${PF}.lock ]; then
+		fuser_pid=$(fuser "${PF}.lock" 2> /dev/null)
```

Comment:
I just noticed that you use `fuser` inside the testbench plumbing and thus in all tests. We cannot do this, basic tests need to be able to run on as plain vanilla systems as possible (we cannot force folks to install everything). So please check if `fuser` is available and downgrade gracefully when not (but otherwise keep the tests running).

For single tests, it is OK if they skip when the tool is not available.

Reply:
Hello @rgerhards - I am finishing changes now that will fall back to old way of getting pid from pidfile when `fuser` command is not available. Would that be ok? My thinking here was that when `fuser` is available, we should use it, because it makes the test stronger.

Reply:
sorry for the late reply - I was swamped with quite some things due to the Corona crisis. I am now re-reviewing.

Yes, if we fall back to the old way, that would be ok. The important thing is that everything continues as before on platforms where fuser is not present (else we will effectively reduce platform coverage).

Reply:
Yes, we fall back to the old way when needed.